src/common/io/socket_address.cpp

```diff
@@ -133,17 +134,17 @@ SocketAddress::SocketResult::Var SocketAddress::socket(const int socket_type) co
         }))
     {
         getLogger("io")->error("Failed to create socket: {}.", std::strerror(err));
-        return err;
+        return static_cast<sdk::ErrorCode>(err);
```

Comment:
For cases like this, we will need a simple mapping function that converts a platform-specific errno into our general ErrorCode.

Static cast to enum results in an unspecified value if the value is not present in the enum definition, so the current approach is not acceptable.

Reply:
> ...  in an unspecified value

I'm not aware of such thing. In contrast, I was sure that underlying value stays as is (regardless of whether it's one of the cases or not). Could you please give me corresponding c++ spec reference?

Reply:
I just checked the standard and I stand corrected -- the following is well-defined:

```c++
enum class Foo : std::uint8_t { bar = 1 };
static_cast<Foo>(2);  // Well-defined, even if there is no field with the value of 2
```

`[expr.static.cast]/10`:

![image](https://github.com/user-attachments/assets/b7636d3e-e3d5-4fb3-a0a2-8f3944712301)

We still have a problem, though. The value of `err` you are getting from an io call may not have a corresponding item in the enumeration, which makes switch statements on the error value problematic. In addition, it violates AUTOSAR A7-2-1 and MISRA C++ 7-2-1.